core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/Guardrail.java

```diff
@@ -0,0 +1,5 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+public interface Guardrail<P extends GuardrailParams, R extends GuardrailResult<R>> {
```

Comment:
Don't you fear clashes with this if we have a bean implementing both input and output guardrails?

Reply:
This indeed makes impossible for a single class to implement both guardrails, and this [latest commit](https://github.com/quarkiverse/quarkus-langchain4j/pull/883/commits/96c09e208df75abda50ded7c24482074b3c909c2) worsen the situation even more, but also makes the API much nicer to be used because now the methods to create the guardrail results are directly available in the respective interface. 

I believe that not allowing a single class to to implement both guardrail interfaces is really a negligible problem (in all honesty I don't think I would like to do this even if possible), but let me know if you think that this is a showstopper instead.

Reply:
I agree and actually rejected that use case in my initial design (see my response to Sergei's comment).

If someone needs to do that, they can just create 3 beans: the bean with the common logic, and the beans with the guards.